compiler/rustc_type_ir/src/lang_items.rs

```diff
@@ -0,0 +1,8 @@
+/// Lang items used by the new trait solver. This can be mapped to whatever internal
+/// representation of `LangItem`s used in the underlying compiler implementation.
+pub enum TraitSolverLangItem {
```

Comment:
This is what I'm doing for now to deal with `LangItem`s. I'm generally okay with it since we're only using a subset of lang items in the solver.